Simulator/Vertices/EventBuffer.h

```diff
@@ -50,17 +50,12 @@ class EventBuffer : public Recordable<uint64_t> {
    /// Get the value of the recordable variable at the specified index.
    /// @param index The index of the recorded value to retrieve.
    /// @return A variant representing the recorded value (uint64_t, double, or string).
-   virtual variant<uint64_t, double, string> getElement(int index) const override;
+   virtual variantTypes getElement(int index) const override;
 
    /// Get A string representing the data type of the recordable variable
-   virtual const string &getDataType() const override;
+   virtual string &getDataType() override;
```

Comment:
A question: this can't be `virtual string &getDataType() const override;`?

Reply:
Currently, the getDataType function is not marked as const because sometimes it changes the data type. I guess the correct approach to updating the value of basicDataType is to create a constructor in each concrete class and call setDataType() within each constructor of the concrete class. I mean, when an object of a concrete class is instantiated, its constructor should set the appropriate data type using setDataType(). This ensures that the basicDataType is updated correctly according to the type of object being created. Is my understanding correct? should I change code by adding constructors and const?

Reply:
This is definitely something that is complex enough that we would need to talk. If you have the time, maybe we can go over the documentation and I can ask stupid questions.

Reply:
Hello Professor, I am happy to have a meeting to go over this. I am available between 9 am and 2 pm, with the exception of Fridays when I am available until 1 pm. If these time slots are not convenient for you, please let me know, as my schedule is quite flexible and I can find an alternative time.

Reply:
From our conversation, this should actually be reverted to the old code, a const method returning a const reference to a strong.